File: test-suite/src/test/java/io/micronaut/docs/client/reactor/MyService.java
package io.micronaut.docs.client.reactor;

//tag::class[]
import io.kubernetes.client.openapi.models.V1PodList;
import io.micronaut.kubernetes.client.reactor.CoreV1ApiReactorClient;
import jakarta.inject.Singleton;
import reactor.core.publisher.Mono;

@Singleton
public class MyService {

    private final CoreV1ApiReactorClient coreV1ApiReactorClient;

    public MyService(CoreV1ApiReactorClient coreV1ApiReactorClient) {
        this.coreV1ApiReactorClient = coreV1ApiReactorClient;
    }

    public void myMethod(String namespace) {
        Mono<V1PodList> v1PodList = coreV1ApiReactorClient.listNamespacedPod(namespace).execute();
    }
}
//end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/client/reactor/MyService.kt
package io.micronaut.docs.client.reactor

//tag::class[]
import io.kubernetes.client.openapi.models.V1PodList
import io.micronaut.kubernetes.client.reactor.CoreV1ApiReactorClient
import jakarta.inject.Singleton
import reactor.core.publisher.Mono

@Singleton
class MyService(private val coreV1ApiReactorClient: CoreV1ApiReactorClient) {

    fun myMethod(namespace: String) {
        val v1PodList: Mono<V1PodList> = coreV1ApiReactorClient.listNamespacedPod(namespace).execute()
    }
}
//end::class[]
